helm/hpcc/templates/_helpers.tpl

```diff
@@ -429,6 +429,85 @@ Pass in dict with root and secretsCategories
 {{- end -}}
 {{- end -}}
 
+{{/*
+Generate Prometheus scrape annotations
+Enables selfdiscovery of metrics service on configured path/port
+Requires sinks[type=prometheus]
+Pass in dict with sinks
+*/}}
+{{- define "hpcc.addPrometheusScrapeAnnotations" -}}
+{{- if hasKey . "sinks" }}
+ {{ range $sink := .sinks -}}
+  {{- if eq (get $sink "type") "prometheus" }}
+   {{- if hasKey $sink "settings" }}
+    {{- if hasKey $sink.settings "autodiscovery" }}
+     {{- if (eq $sink.settings.autodiscovery true ) }}
+prometheus.io/scrape: 'true'
+prometheus.io/path: {{ $sink.settings.path | default "/metrics"}}
+prometheus.io/port: {{  $sink.settings.port | default 8767 | quote }}
+     {{ end }}
+    {{ end }}
+   {{ end }}
+  {{ end }}
+ {{ end}}
+{{ end}}
+{{- end -}}
+
+
+{{/*
+Generate NetworkPolicy exposing if global.metrics.sink[type=prometheus].settins.port differs from
```

Comment:
Odd function... the concept is, a networkpolicy is created for global.metrics.sinks[type=prometheus].settings.port.
If <component>.metrics.sinks[type=prometheus].settings.port exists and differs, create dedicated networkpolicy. The podSelector likely should change.